src/TriggerUnit.cpp

```diff
@@ -451,18 +439,18 @@ QString TriggerUnit::assembleReport()
 
 void TriggerUnit::doCleanup()
 {
-    for(auto it = mCleanupList.begin(); it != mCleanupList.end(); it++)
+    for(auto & trigger : mCleanupList)
```

Comment:
You are quite right good sir, a reference to a pointer is a bit redundant.

btw learnt this useful trick to figure out what ``auto`` resolves to:

```cpp
template <typename T> struct watzattype;
void TriggerUnit::doCleanup()
{
    for(auto & trigger : mCleanupList)
    {
        watzattype<decltype(trigger)>{};
```

Compiler will error and tell you the type, ``TTrigger*&`` in this case haha.